### PR TITLE
Sighash support for BIP341/342/118

### PR DESCRIPTION
**Pull request by instagibbs**
This has received some light testing as part of my work on eltoo in CLN. Should be very simple to add the actual signature support on top, which I've done in my local CLN repo in a few lines.

Partially resolves https://github.com/ElementsProject/libwally-core/issues/304

cc @greenaddress

### CODE REVIEW

**Comment by jgriffiths** [comment]
Hi @instagibbs thanks for this!

Will need a little tweaking for merging, I've merged the first commit to https://github.com/ElementsProject/libwally-core/pull/351 and will work on the others as time permits. Leaving this open to track.

**Comment by instagibbs** [comment]
Good stuff, let me know whatever is needed

File: src/transaction.c [hunk 1]
```diff
@@ -2089,6 +2100,273 @@ static inline int tx_to_bip143_bytes(const struct wally_tx *tx,
...
+        /* amount (8) */
+        p += uint64_to_le_bytes(opts->satoshi, p);
+        /* scriptPubKey (35) */
+        if (opts->prev_script_lens[opts->index] != 34) {
+            goto error;
+        }
+        p += varbuff_to_bytes(opts->prev_scripts[opts->index], opts->prev_script_lens[opts->index], p);
+        /* nSequence (4) */
+        p += uint32_to_le_bytes(tx->inputs[opts->index].sequence, p);
+    } else if (sh_apoas) {
```
Comment (jgriffiths) [inline src/transaction.c:2306]: 2

**Comment by jgriffiths** [comment]
@instagibbs as discussed offline, I've pushed the reworked commits to https://github.com/ElementsProject/libwally-core/pull/351 if you'd like to review there (or offline). Thanks!

**Pull request closed without merging**